### DISCUSSION TITLE
i think theres an issue with typescript on my end any help would be appreciated

### DISCUSSION THREAD

**Discussion by valterebizh** [jef/streetmerchant#3041] [Help] [answered]
when i run npm run start i get this error output:

Truncated to just show end:
..........

test/test-blank.ts:1:25 - error TS2307: Cannot find module 'assert' or its corresponding type declarations.

1 import * as assert from 'assert';
                          ~~~~~~~~

test/util.ts:1:31 - error TS2307: Cannot find module 'puppeteer' or its corresponding type declarations.

1 import {Browser, launch} from 'puppeteer';
                                ~~~~~~~~~~~

test/util.ts:59:44 - error TS2705: An async function or method in ES5/ES3 requires the 'Promise' constructor.  Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your '--lib' option.

59 export async function launchTestBrowser(): Promise<Browser> 

Found 336 errors in 44 files.

Errors  Files
     4  src/adblocker.ts:1
     3  src/banner.ts:1
   177  src/config.ts:1
    11  src/index.ts:1
     2  src/logger.ts:2
     1  src/messaging/apns.ts:3
     1  src/messaging/captcha.ts:21
     4  src/messaging/desktop.ts:4
    10  src/messaging/discord.ts:2
     2  src/messaging/email.ts:3
     3  src/messaging/freemobile.ts:4
     3  src/messaging/gotify.ts:4
     2  src/messaging/mqtt.ts:2
     4  src/messaging/philips-hue.ts:1
     1  src/messaging/redis.ts:2
    10  src/messaging/slack.ts:1
     1  src/messaging/sms.ts:3
     1  src/messaging/sound.ts:3
     3  src/messaging/streamlabs.ts:4
     3  src/messaging/telegram.ts:3
     3  src/messaging/twilio.ts:3
     3  src/messaging/twitch.ts:3
     1  src/messaging/twitter.ts:3
     4  src/store/captcha-handler.ts:1
     6  src/store/fetch-links.ts:3
     2  src/store/filter.ts:14
     6  src/store/includes-labels.ts:2
    20  src/store/lookup.ts:8
     1  src/store/model/asus.ts:2
     1  src/store/model/helpers/backoff.ts:17
     1  src/store/model/helpers/card.ts:20
     9  src/store/model/index.ts:169
     2  src/store/model/microcenter.ts:7
     1  src/store/model/newegg-ca.ts:2
     1  src/store/model/newegg-sg.ts:2
     1  src/store/model/newegg.ts:2
     1  src/store/model/playstation.ts:2
     2  src/store/model/store.ts:1
    11  src/util.ts:1
     7  src/web/index.ts:1
     1  test/functional/test-captcha.ts:8
     1  test/functional/test-notification.ts:1
     1  test/test-blank.ts:1
     2  test/util.ts:1
     
     
     Let me know what i am doing wrong, i had this working on my last mac but on the new mac its giving me issues.

**Comment by Mr-Tech-13** [comment]
Have you ran npm install?
Are you in the right directory?

**Reply by valterebizh** [reply]
yep

**Reply by valterebizh** [reply]
I wonder if this has something to do with my new M2 mac.... seems like issue with running npm on arm64... but all looked good:

streetmerchant git:(main) npm -v
8.19.2
➜  streetmerchant git:(main) node -p process.arch
arm64

If you can help troubleshoot thats nice if not dont worry about it! this is so 2020 :) and hopefully the great GPU shortage is behind us.

**Comment by valterebizh** [comment, marked as answer]
all good recompiled to run in rosetta issue is with node and arm architecture
